test/integration/isolated/udpingress_test.go

```diff
@@ -0,0 +1,156 @@
+		New("essentials").
+		WithLabel(testlabels.NetworkingFamily, testlabels.NetworkingFamilyGatewayAPI).
+		WithLabel(testlabels.Kind, testlabels.KindKongUDPIngress).
+		Setup(SkipIfRouterNotExpressions).
+		WithSetup("deploy kong addon into cluster", featureSetup()).
+		WithSetup("configure a udpecho Deployment and Service", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			// kongClient, err := gatewayclient.NewForConfig(cfg.Client().RESTConfig())
+			kongClient, err := clientset.NewForConfig(cfg.Client().RESTConfig())
+			assert.NoError(t, err)
+			ctx = SetInCtxForT(ctx, t, kongClient)
+
+			cleaner := GetFromCtxForT[*clusters.Cleaner](ctx, t)
+			namespace := GetNamespaceForT(ctx, t)
+			cluster := GetClusterFromCtx(ctx)
+
+			t.Log("creating a udpecho pod to test UDPRoute traffic routing")
+			container := generators.NewContainer(serviceName, test.EchoImage, test.EchoUDPPort)
+			// App go-echo sends a "Running on Pod <UUID>." immediately on connecting.
+			container.Env = []corev1.EnvVar{
+				{
+					Name:  "POD_NAME",
+					Value: testUUID,
+				},
+			}
+			deployment := generators.NewDeploymentForContainer(container)
+			deployment, err = cluster.Client().AppsV1().Deployments(namespace).Create(ctx, deployment, metav1.CreateOptions{})
+			assert.NoError(t, err)
+			cleaner.Add(deployment)
+
+			t.Logf("exposing deployment %s/%s via service", deployment.Namespace, deployment.Name)
+			service := generators.NewServiceForDeployment(deployment, corev1.ServiceTypeClusterIP)
+			service.Name = serviceName
+			// Use the same port as the default UDP port from the Kong Gateway deployment
+			// to the udpecho port, as this is what will be used to route the traffic at the Gateway.
+			service.Spec.Ports = []corev1.ServicePort{{
+				Name:       "udp",
+				Protocol:   corev1.ProtocolUDP,
+				Port:       servicePort,
+				TargetPort: intstr.FromInt(test.EchoUDPPort),
+			}}
+			service, err = cluster.Client().CoreV1().Services(namespace).Create(ctx, service, metav1.CreateOptions{})
+			assert.NoError(t, err)
+			cleaner.Add(service)
+
+			t.Logf("creating a UDPIngress to access deployment %s via kong", deployment.Name)
+			udpIngress := &kongv1beta1.UDPIngress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: udpIngressName,
+					Annotations: map[string]string{
+						annotations.IngressClassKey: GetIngressClassFromCtx(ctx),
+					},
+				},
+				Spec: kongv1beta1.UDPIngressSpec{Rules: []kongv1beta1.UDPIngressRule{
+					{
+						Port: ktfkong.DefaultUDPServicePort,
+						Backend: kongv1beta1.IngressBackend{
+							ServiceName: service.Name,
+							ServicePort: servicePort,
+						},
+					},
+				}},
+			}
+			_, err = kongClient.ConfigurationV1beta1().UDPIngresses(namespace).Create(ctx, udpIngress, metav1.CreateOptions{})
+			assert.NoError(t, err)
+
+			return ctx
+		}).
+		Assess("basic test - UDPIngress status and connectivity", func(ctx context.Context, t *testing.T, _ *envconf.Config) context.Context {
```

Comment:
```suggestion
		Assess("basic test - status and connectivity", func(ctx context.Context, t *testing.T, _ *envconf.Config) context.Context {
```

The `UDPIngress` is already mentioned in the test name, isn't it?